pineappl_py/src/evolution.rs

```diff
@@ -44,3 +44,9 @@ impl PyEvolveInfo {
 pub struct PyOperatorSliceInfo {
     pub(crate) slice_info: OperatorSliceInfo,
 }
+
+#[pymodule]
+pub fn evolution(_py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_class::<PyEvolveInfo>()?;
```

Comment:
- we need to expose `PyOperatorSliceInfo` for https://github.com/NNPDF/pineko/pull/181 
- although `pineappl::EvolveInfo` is not officially deprecated, I think it practice it is (i.e. I could not find a relevant use case) - so I suggest to drop it (even maybe in the main library)
- as we concluded that the naming scheme with `name` in the macro is the recommended way, we should propagate that through out - i.e. `PyOperatorSliceInfo` should be exposed as `OperatorSliceInfo` - in sum pineko should change from `from pineappl.grid import PyOperatorSliceInfo` to `from pineappl.evolution import OperatorSliceInfo`

Reply:
wait a second - I think this module should just be dropped as is: `PyOperatorSliceInfo` is correctly exposed in `grid.rs`

Reply:
Yes, `OperatorSliceInfo` is correctly exposed and `pineko` will be calling it as `from pineappl.grid import OperatorSliceInfo`.

> although pineappl::EvolveInfo is not officially deprecated, I think it practice it is (i.e. I could not find a relevant use case) - so I suggest to drop it (even maybe in the main library)

Just to confirm, you only want to remove the exposed `pineappl.evolution` module right? Because we cannot completely drop `evolution.rs` as it is used here:
https://github.com/NNPDF/pineappl/blob/93d3c634d8df92d6e310bc79d7d75e9e711d7054/pineappl_py/src/grid.rs#L607

and subsequently in `pineko`.

Reply:
sorry, I was confused - of course `pineko` needs the module (I had the misconception that `EvolveInfo` and `OperatorSliceInfo` are the same - which of course they are not)